src/opensles/EngineOpenSLES.cpp

```diff
@@ -30,9 +62,11 @@ SLresult EngineOpenSLES::open() {
 
     SLresult result = SL_RESULT_SUCCESS;
     if (mOpenCount++ == 0) {
+        // load the library and link to it
+        loadOpenSLES();
 
         // create engine
-        result = slCreateEngine(&mEngineObject, 0, NULL, 0, NULL, NULL);
+        result = (*gFunction_slCreateEngine)(&mEngineObject, 0, NULL, 0, NULL, NULL);
```

Comment:
Should we check whether gFunction_slCreateEngine is null first?

Reply:
And in the case whether it is null, what should be our backup? Can we alert the user somehow with a notification?

Reply:
I check the link result so now it is guaranteed to be non-null.